File: Baekjoon/수학/9020.py
# https://www.acmicpc.net/problem/9020

import sys
input = sys.stdin.readline

t = int(input())

for _ in range(t):
    n = int(input())
    sieve = [False, False] + [True] * (n - 1)

    for i in range(2, int(n ** 0.5) + 1):
        if sieve[i]:
            for j in range(2 * i, n + 1, i):
                sieve[j] = False

    for i in range(n // 2, 1, -1):
        if sieve[i] and sieve[n - i]:
            print(i, n - i)
            break

File: Baekjoon/수학/9020.cpp
// https://www.acmicpc.net/problem/9020

#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

int main() {
    vector<bool> prime(10001, true);
    prime[0] = prime[1] = false;
    for (int i = 2; i <= sqrt(10000); i++) {
        if (prime[i]) {
            for (int j = i * i; j <= 10000; j += i) {
                prime[j] = false;
            }
        }
    }
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        for (int i = n / 2; i >= 2; i--) {
            if (prime[i] && prime[n - i]) {
                cout << i << ' ' << n - i << '\n';
                break;
            }
        }
    }
    return 0;
}